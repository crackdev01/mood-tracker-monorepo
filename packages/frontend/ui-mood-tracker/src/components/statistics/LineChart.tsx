import React from 'react';
import Chart from 'react-apexcharts';
import { useTranslation } from 'react-i18next';

const LineChart = (props: any) => {
  const { t } = useTranslation(['Statistics']);
  const series = [
    {
      name: 'Intensity',
      data: props.data,
    },
  ];
  const options = {
    chart: {
      height: 350,
      type: 'line',
      zoom: {
        enabled: true,
      },
    },
    dataLabels: {
      enabled: true,
    },
    stroke: {
      width: 2,
      curve: 'smooth',
    },
    yaxis: [
      {
        title: {
          text: t('chart.yaxis'),
        },
      },
    ],
    colors: ['#210124'],
    fill: {
      type: 'gradient',
      gradient: {
        shadeIntensity: 1,
        inverseColors: true,
        gradientToColors: ['#1E3A63'],
        opacityFrom: 1,
        opacityTo: 1,
        type: 'vertical',
        stops: [0, 30],
      },
    },
    tooltip: {
      shared: true,
      intersect: false,
      y: {
        formatter: function (y: any) {
          if (typeof y !== 'undefined') {
            return 'Intensity Value: ' + y.toFixed(0);
          }
          return y;
        },
      },
    },
  };
  return (
    <div id="chart">
      <Chart options={options} series={series} type="line" height={350} width="100%" />
    </div>
  );
};

export default LineChart;
