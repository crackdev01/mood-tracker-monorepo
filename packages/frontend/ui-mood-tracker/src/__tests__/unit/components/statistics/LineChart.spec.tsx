import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import LineChart from '../../../../components/statistics/LineChart';
import { mockMoods } from '../../../fixtures';

Enzyme.configure({ adapter: new Adapter() });

describe('LineChart', () => {
  test('renders default', () => {
    const props = {
      data: mockMoods
        .map((m: any) => {
          return {
            x: m.mood_enteredAt,
            y: m.mood_intensity,
          };
        }
      ),
    };
    const w = shallow(<LineChart {...props} />);
    expect(w.debug()).toMatchSnapshot();
  });
});
