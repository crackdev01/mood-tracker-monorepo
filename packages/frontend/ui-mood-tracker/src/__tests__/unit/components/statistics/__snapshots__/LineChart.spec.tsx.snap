// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart renders default 1`] = `
"<div id=\\"chart\\">
  <r options={{...}} series={{...}} type=\\"line\\" height={350} width=\\"100%\\" />
</div>"
`;
